Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -621,9 +624,19 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
                 // Add to document map
                 documentLocationMap[url] = technologyResult.topicGraphNode.reference
                 
-                let sourceLanguage = SourceLanguage.swift
-                let node = DocumentationNode(reference: technologyResult.topicGraphNode.reference, kind: .technology, sourceLanguage: sourceLanguage, name: .conceptual(title: technology.intro.title), markup: technology.originalMarkup, semantic: technology)
-                documentationCache[technologyResult.topicGraphNode.reference] = node
+                let reference = technologyResult.topicGraphNode.reference.withSourceLanguages(sourceLanguages)
+                
+                let node = DocumentationNode(
+                    reference: reference,
+                    kind: .technology,
+                    sourceLanguage: Self.defaultLanguage(in: sourceLanguages),
+                    availableSourceLanguages: sourceLanguages,
+                    name: .conceptual(title: technology.intro.title),
+                    markup: technology.originalMarkup,
+                    semantic: technology
+                )
+                documentationCache[reference] = node
+                topicGraph.nodes[reference]?.reference = reference
```

Comment:
Is this a change from before? We weren't including technology references in the topic graph?

Reply:
We are at an earlier stage, in `registerDocuments`. What we're doing here is updating the reference now that we have enough information to know in which languages the page is available in. I'll add a comment to clarify that.